services/wallet/thirdparty/rarible/client_test.go

```diff
@@ -0,0 +1,345 @@
+package rarible
+
+import (
+	"encoding/json"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/status-im/status-go/services/wallet/bigint"
+	"github.com/status-im/status-go/services/wallet/thirdparty"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnmarshallCollection(t *testing.T) {
+	collectionJSON := `{
+		"id": "ETHEREUM:0x06012c8cf97bead5deae237070f9587f8e7a266d",
+		"blockchain": "ETHEREUM",
+		"structure": "REGULAR",
+		"type": "ERC721",
+		"status": "CONFIRMED",
+		"name": "CryptoKitties",
+		"symbol": "CK",
+		"features": [],
+		"minters": [],
+		"meta": {
+		  "name": "CryptoKitties",
+		  "description": "CryptoKitties is a game centered around breedable, collectible, and oh-so-adorable creatures we call CryptoKitties! Each cat is one-of-a-kind and 100% owned by you; it cannot be replicated, taken away, or destroyed.",
+		  "tags": [],
+		  "genres": [],
+		  "externalUri": "https://www.cryptokitties.co/",
+		  "content": [
+			{
+			  "@type": "IMAGE",
+			  "url": "https://i.seadn.io/gae/C272ZRW1RGGef9vKMePFSCeKc1Lw6U40wl9ofNVxzUxFdj84hH9xJRQNf-7wgs7W8qw8RWe-1ybKp-VKuU5D-tg?w=500&auto=format",
+			  "representation": "ORIGINAL",
+			  "mimeType": "image/png",
+			  "size": 7584,
+			  "available": true,
+			  "width": 250,
+			  "height": 246
+			}
+		  ],
+		  "externalLink": "https://www.cryptokitties.co/",
+		  "sellerFeeBasisPoints": 250
+		},
+		"originOrders": [],
+		"self": false,
+		"scam": false
+	  }`
+	expectedCollectionData := thirdparty.CollectionData{
+		ID: thirdparty.ContractID{
+			ChainID: 1,
+			Address: common.HexToAddress("0x06012c8cf97bead5deae237070f9587f8e7a266d"),
+		},
+		Provider: "rarible",
+		Name:     "CryptoKitties",
+		ImageURL: "https://i.seadn.io/gae/C272ZRW1RGGef9vKMePFSCeKc1Lw6U40wl9ofNVxzUxFdj84hH9xJRQNf-7wgs7W8qw8RWe-1ybKp-VKuU5D-tg?w=500&auto=format",
+		Traits:   make(map[string]thirdparty.CollectionTrait),
+	}
+
+	collection := Collection{}
+	err := json.Unmarshal([]byte(collectionJSON), &collection)
+	assert.NoError(t, err)
+
+	contractID, err := raribleContractIDToUniqueID(collection.ID, true)
+	assert.NoError(t, err)
+
+	collectionData := collection.toCommon(contractID)
+	assert.Equal(t, expectedCollectionData, collectionData)
+}
+
+func TestUnmarshallOwnedCollectibles(t *testing.T) {
+	ownedCollectiblesJSON := `{
+		"continuation": "1603836283000_ETHEREUM:0xd07dc4262bcdbf85190c01c996b4c06a461d2430:63879:0x4765273c477c2dc484da4f1984639e943adccfeb",
```

Comment:
What do you think about moving these massive response data into a separate test_data file? It might improve code readability.

Reply:
done